src/Application/GameStates/VideoState.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <Library/Fsm/FsmState.h>
+#include <GUI/GUIEnums.h>
+
+#include <string>
+#include <memory>
+
+class IMovie;
+
+class VideoState : public FsmState {
+ public:
+    explicit VideoState(std::string_view videoFileName, std::string_view videoGroup);
+    virtual FsmAction update() override;
+    virtual FsmAction enter() override;
+    virtual void exit() override;
+
+    static const std::string_view LOGO_GROUP;
+    static const std::string_view INTRO_GROUP;
```

Comment:
Is there a reason why these are strings and not type-safe enums? Realistically, we won't be passing anything else into `VideoState` ctor, and using an enum clearly communicates what the ctor arg is about, unlike with a string arg.

This is about "code that's pretty much self-documented" VS "code that needs additional docs to figure out."

Reply:
I think I have this tendency into thinking about the modding aspect of it.

This is a broader discussion related to each enumerations we have in game. And probably you have some ideas in mind already on how to tackle that.
If I add an enumeration for the video groups/types and from lua I'd like to add a new video state and I want that state to belong to a new group then I can't. Or to be better, I can't with a simple enumeration; there are of course different approaches we could use to "expand things from scripting".
Probably we'll have to go through this anyway in the future and maybe trying to use strings to "solve" a potential, and very localized, use case like this is not ok.

I'm going to change the strings to enums as you said. It's more predictable and most probably stability is way more important than flexibility ( take those adjective with a grain of salt anyway 😅) at this stage.